lib/heroku/kensa/check.rb

```diff
@@ -290,6 +298,73 @@ def call!
           reader, writer = IO.pipe
         end
 
+        if data[:async]
+          child = fork do
+            reader.close
+            server = TCPServer.open(7779)
+            client = server.accept
+            writer.write(client.readpartial(READLEN))
+            client.write("HTTP/1.1 200 OK\r\nContent-Length: 0\r\n\r\n")
+            client.close
+            writer.close
+          end
+          sleep(1)
+        end
+
+        code1, json1 = post(credentials, base_path, payload)
+        code2, json2 = post(credentials, base_path, payload)
+
+        json1 = OkJson.decode(json1)
+        json2 = OkJson.decode(json2)
+
+        if data["api"].fetch("requires", []).include?("many_per_app")
+          check "returns different ids" do
+            if json1["id"] == json2["id"]
+              error "multiple provisions cannot return the same id"
+            else
+              true
+            end
+          end
+        else
+          check "disallows duplicate provisions" do
+            if code2 == 200
+              error("should not allow duplicate provisions")
```

Comment:
Not idempotent (that would imply returning the same provision/account details) but it shouldn't be an error condition. They should always provision as many add-ons as requests we make, under the premise that the first one may be pending delete. 

There's a counter argument here though: when receiving a duplicate request, they could check the App Info API to see if the other one is still installed and reject on that premise. However, this test might encourage people to too aggressively reject valid installations.